contracts/finality/src/utils.rs

```diff
@@ -43,6 +44,35 @@ pub fn query_finality_provider(
     Ok(res)
 }
 
+/// Validates that a consumer ID has a valid format (non-empty, valid characters, etc.)
+pub fn validate_consumer_id_format(consumer_id: &str) -> Result<(), ContractError> {
+    if consumer_id.is_empty() {
+        return Err(ContractError::InvalidConsumerId(
+            "Consumer ID cannot be empty".to_string(),
+        ));
+    }
+
+    // Check for valid characters (alphanumeric, hyphens, underscores)
+    if !consumer_id
+        .chars()
+        .all(|c| c.is_alphanumeric() || c == '-' || c == '_')
+    {
+        return Err(ContractError::InvalidConsumerId(
+            "Consumer ID can only contain alphanumeric characters, hyphens, and underscores"
+                .to_string(),
+        ));
+    }
+
+    // Check length (reasonable bounds)
+    if consumer_id.len() > 100 {
+        return Err(ContractError::InvalidConsumerId(
+            "Consumer ID cannot exceed 100 characters".to_string(),
```

Comment:
@KonradStaniec Should we enforce these validation rules in Babylon node as well? Those validations seem very reasonable